src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotDeclareVisibleInstanceFields.cs

```diff
@@ -24,15 +25,25 @@ public abstract class DoNotDeclareVisibleInstanceFieldsAnalyzer : DiagnosticAnal
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182141.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterSymbolAction(obj =>
+            {
+                var field = (IFieldSymbol)obj.Symbol;
+
+                // Only report diagnostic on non-static, non-const, public fields
+                if (!field.IsStatic && !field.IsConst && field.IsPublic())
```

Comment:
Looking at the FxCop implementation of this rule, it only fires when the field is externally visible (i.e field.GetResultantVisilibity == Public) and it also has a few exclusion cases - it doesnt fire in winmds or in a type that derives from "System.Web.UI.Control" or if the field is only needed for interop purposes.